Lib/zipfile/__init__.py

```diff
@@ -331,7 +331,7 @@ def _EndRecData(fpin):
     # number does not appear in the comment.
     maxCommentStart = max(filesize - (1 << 16) - sizeEndCentDir, 0)
     fpin.seek(maxCommentStart, 0)
-    data = fpin.read()
+    data = fpin.read(ZIP_MAX_COMMENT + sizeEndCentDir + 1)
```

Comment:
Maybe add an assert checking that you are indeed at the end of the file? (just in case).

Reply:
I don't see a good way to test that locally. In theory the seek just before this calculated that out and moved so that this read size will always read past the end. `test_zipfile` does have a test for appending a comment which exercises this